huggingface_pipelines/text.py

```diff
@@ -0,0 +1,518 @@
+        self.config = config
+        self.t2vec_model = TextToEmbeddingModelPipeline(
+            encoder=self.config.encoder_model,
+            tokenizer=self.config.encoder_model,
+            device=self.config.device,
+        )
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Process a batch of texts and convert them to embeddings.
+
+        Args:
+            batch (Dict[str, Any]): Input batch containing texts.
+
+        Returns:
+            Dict[str, Any]: Processed batch with encoded embeddings.
+        """
+        for column in self.config.columns:
+            if column in batch:
+                assert all(
+                    isinstance(item, list) for item in batch[column]
+                ), f"Column {column} must contain only lists of sentences, not individual strings."
+
+                all_sentences = [
+                    sentence for item in batch[column] for sentence in item
+                ]
+                all_embeddings = self.encode_texts(all_sentences)
+
+                sentence_embeddings = []
+                start_idx = 0
+                for item in batch[column]:
+                    end_idx = start_idx + len(item)
+                    sentence_embeddings.append(all_embeddings[start_idx:end_idx])
+                    start_idx = end_idx
+
+                batch[f"{column}_{self.config.output_column_suffix}"] = (
+                    sentence_embeddings
+                )
+                logger.debug(f"{column} column embeddings: {batch[column][:5]}")
+            else:
+                logger.warning(f"Column {column} not found in batch.")
+
+        return batch
+
+    def encode_texts(self, texts: List[str]) -> np.ndarray:
+        """
+        Encode a list of texts into embeddings.
+
+        Args:
+            texts (List[str]): List of texts to encode.
+
+        Returns:
+            np.ndarray: Array of embeddings.
+
+        Raises:
+            Exception: If there's an error during encoding.
+        """
+        try:
+            embeddings: List[torch.Tensor] = []
+            for i in range(0, len(texts), self.config.batch_size):
+                batch_texts = texts[i : i + self.config.batch_size]
+                batch_embeddings = self.t2vec_model.predict(
+                    batch_texts,
+                    source_lang=self.config.source_lang,
+                    batch_size=self.config.batch_size,
+                    max_seq_len=self.config.max_seq_len,
+                )
+
+                batch_embeddings = batch_embeddings.detach().cpu().numpy()
+                embeddings.extend(batch_embeddings)
+
+            # We return this as np array for more efficient memory representation
+            return np.array(embeddings)
+        except Exception as e:
+            logger.error(f"Error encoding texts or sentences: {e}")
+            raise
+
+
+class TextToEmbeddingPipelineFactory(PipelineFactory):
```

Comment:
instead of redefining create_pipeline each time, you can just use some class templates with config and pipeline class as param